lib/Backend/arm/EncoderMD.cpp

```diff
@@ -2241,7 +2241,7 @@ EncoderMD::BaseAndOffsetFromSym(IR::SymOpnd *symOpnd, RegNum *pBaseReg, int32 *p
         if (func->GetMaxInlineeArgOutSize() != 0)
         {
             Assert(func->HasInlinee());
-            Assert(baseReg == RegSP);
+            Assert(baseReg == RegSP || (func->HasTry() && baseReg == RegR7));
```

Comment:
If the function has a try, but the `baseReg == RegSP` is that ok, or is that also an error? e.g. should the `baseReg` be `RegR7` if and only if the function has a try?

Reply:
The baseReg is always R7 when there is a try.

Reply:
In that case, I suggest something like `Assert(baseReg == (func->HasTry() ? RegR7 : RegSP))` to enforce that